mewarpx/mewarpx/utils_store/util.py

```diff
@@ -387,3 +388,35 @@ def plasma_Debye_length(T, n):
     return np.sqrt(
         constants.epsilon_0 * constants.kb_J * T / (n * constants.e**2)
     )
+
+def calc_timesteps(V_anode, V_grid, CFL_factor, nx, nz, xmin, xmax, zmin, zmax):
+        """Calculate timestep sizes and crossing timestep number.
+
+         Arguments:
+            V_anode (float): Vacuum bias of anode relative to cathode, V.
+            V_grid (float): Vaccum bias of highest-voltage grid relative to
+                cathode, V.
+            CFL_factor (float): Multiplier to determine the actual timestep
+                given the CFL ratio. eg. ``dt = CFL_factor * CFL_ratio``.
+            nx (int): Number of cells in the x direction
+            nz (int): Number of cells in the z direction
+            xmin (float): The minimum potential in the x direction
+            xmax (float): The maxminum potential in the x direction
+            zmin (float): The minimum potential in the z direction
+            zmax (float): The maximum potential in the z direction
+        """
+        vmax = np.sqrt(
+               2*np.e/constants.m_e * max(0.0, V_anode, V_grid)
+        )
+
+        if nx > 0:
+            dx = (xmax - xmin) / nx
+        else:
+            dx = 0.
+        dz = (zmax - zmin) / nz
+
+        if dx > 0.:
+            dt_local = min(dx, dz) / vmax * CFL_factor
+        else:
+            dt_local = dz / vmax * CFL_factor
+        return parallel_util.mpiallreduce(data=dt_local, op=MPI.MIN)
```

Comment:
To save on having to import MPI for util, we could change `mpiallreduce` so that it takes in a string as an argument for `op`. For example 
```

def mpiallreduce(data=None, opstring=None, comm=None):
    if opstring is None:
        op = mpi.SUM
    elif opstring == "SUM":
        op = mpi.SUM:
    elif opstring == "MIN":
        op = MPI.MIN
    else:
        raise NotImplementedError("The opstring is unrecognized or has not been implemented yet.")
    if comm is None:
        comm = comm_world
    # --- "fast" version was removed because it produced bugs
    if isinstance(data, np.ndarray) and data.dtype is not np.dtype('object'):
        result = np.empty_like(data)
        comm.Allreduce(data, result, op=op)
    else:
result = comm.allreduce(data, op=op)
```

Reply:
Is there a negative performance impact or some other reason we want to avoid importing MPI?

Reply:
Overall I don't think it wouldn't be that big of an impact since the import is just being called once. But this does mean that runs that don't use mpi4py, would still have mpi4py cached as it is imported in this file.

Reply:
Any thoughts on this change @roelof-groenewald ?

Reply:
> Overall I don't think it wouldn't be that big of an impact since the import is just being called once. But this does mean that runs that don't use mpi4py, would still have mpi4py cached as it is imported in this file.

Yeah that's a good point

Reply:
I think not needing to import mpi4py for non-parallel runs would be good so I'd say lets go with Kevin's suggestion.